dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/kafka/framework/AbstractReceptionBean.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.reactive.messaging.fat.kafka.framework;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import org.eclipse.microprofile.reactive.messaging.Message;
+
+/**
+ * Test bean to monitor reception of messages from reactive messaging
+ * <p>
+ * To use this class, subclass it and override {@link #recieveMessage(Message)} to add {@code @Incoming} and {@code @Acknowledgement} annotations.
+ */
+public class AbstractReceptionBean {
+
+    private final Queue<Message<String>> receivedMessages = new LinkedList<>();
+
+    public CompletionStage<Void> recieveMessage(Message<String> msg) {
+        synchronized (this) {
+            receivedMessages.add(msg);
+            this.notifyAll();
+        }
+        return CompletableFuture.completedFuture(null);
+    }
+
+    public List<Message<String>> getReceivedMessages(int count, Duration timeout) throws InterruptedException {
+        ArrayList<Message<String>> result = new ArrayList<>();
+        Duration remaining = timeout;
+        long startTime = System.nanoTime();
+        while (!remaining.isNegative() && result.size() < count) {
+            synchronized (this) {
+                while (!receivedMessages.isEmpty() && result.size() < count) {
+                    result.add(receivedMessages.poll());
+                }
+                if (result.size() < count) {
+                    this.wait(remaining.toMillis());
```

Comment:
Yes there is. This is a call to `Object.wait()` rather than `Thread.sleep()` and the interruption comes from the call to `notifyAll()` on line 38 when a new message is received.

The flow should be:
1. Grab up to `count` messages
2. If not, wait for up to `timeout`
3. If new messages are delivered, grab them and go back to waiting
4. Stop waiting if the total wait time reaches `timeout`
5. Assert then return